src/group_map.rs

```diff
@@ -768,8 +767,13 @@ where
     /// assert!(*val == 1 || *val == 2);
     /// assert!(iter.next().is_none());
     /// ```
-    pub fn get_left_iter(&self, item: &R) -> Option<PairIter<'_, L, R, S>> {
-        let r_hash = make_hash::<R, S>(&self.hash_builder, item);
+    pub fn get_left_iter<Q>(&self, item: &Q) -> Option<PairIter<'_, L, R, S>>
+    where
+        R: Borrow<Q>,
+        Q: Hash + Eq,
+        RightItem<R>: PartialEq<Q>,
```

Comment:
I wonder if we can remove this bound and my the `LeftItem` and `RightItem` struct private if we implement `PartialEq<Q>` for the item structs. This would look something like this:
```rust
impl<R, Q> PartialEq<Q> for RightItem<R>
where R: Borrow<Q>,
      Q: Hash + Eq, { ... }
```

Reply:
it becomes a conflicting implementation

I was thinking it was because `RightItem` itself implements `Hash + Eq` and `Borrow` has a blanket implementation `impl<T> Borrow<T> for T`, but breaking down to a simpler case the problem persists: https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=a1abc1e50d22e3503ee8ce56d0af44e6

I've found a different, less elegant, solution